pkg/cli/cmd.go

```diff
@@ -142,15 +145,41 @@ type MetricsBuilder interface {
 	Build() (io.Reader, error)
 }
 
-func (t Telemetry) CustomMetricsHandlerFor(mb MetricsBuilder) http.Handler {
+func (t Telemetry) CustomMetricsHandlerFor(reg *prometheus.Registry, mb MetricsBuilder) http.Handler {
 	if t.Flags.Metrics {
```

Comment:
nit: 
``` golang
if !t.Flags.Metrics {
        var empty http.HandlerFunc = func(w http.ResponseWriter, r *http.Request) {
		w.Write([]byte(""))
	}
	return empty
}

Reply:
I get your point. 👍  Lets keep it as is to unify with others for now, will address it by another PR (apply this pattern to others metrics handler as well)

Reply:
Changed my mind, will address it in this PR and apply only for this handler, lol

Reply:
Addressed by https://github.com/pipe-cd/pipe/pull/2279/commits/6513a2d87a265e2d656c8514a66d3ee2d8782886 🙏